src/guiapi/include/WindowMenuItems.hpp

```diff
@@ -146,12 +196,65 @@ class WI_SWITCH_t : public AddSuper<IWiSwitch> {
     //cannot create const std::array<const char *, SZ> with std::initializer_list<const char*>
     //template<class ...E> and {{std::forward<E>(e)...}} is workaround
     template <class... E>
-    WI_SWITCH_t(int32_t index, string_view_utf8 label, uint16_t id_icon, bool enabled, bool hidden, E &&... e)
+    WI_SWITCH_t(int32_t index, string_view_utf8 label, uint16_t id_icon, is_enabled_t enabled, is_hidden_t hidden, E &&... e)
         : AddSuper<IWiSwitch>(index, label, id_icon, enabled, hidden)
         , items { { std::forward<E>(e)... } } {}
 
+    virtual void InitRollIfNeeded(IWindowMenu &window_menu, Rect16 rect) override {
+        reInitRoll(window_menu, getSwitchMenuRects(window_menu, rect)[0]);
+    }
+
 protected:
     virtual const char *get_item() const override { return items[index]; }
+
+    /**
+    * Returns Menu item rect devided into 4 rects according to this format: [label], ['['], [switch], [']']
+    **/
+    std::array<Rect16, 4> getSwitchMenuRects(IWindowMenu &window_menu, Rect16 rect) const {
+        Rect16 base_rect = getRollingRect(window_menu, rect);
+        Rect16 switch_rect, bracket_start, bracket_end;
+
+        bracket_end = { int16_t(base_rect.Left() + base_rect.Width() - window_menu.secondary_font->w), base_rect.Top(), uint16_t(window_menu.secondary_font->w), base_rect.Height() };
+        base_rect -= bracket_end.Width();
+
+        string_view_utf8 localizedItem = _(get_item());
+        switch_rect = getCustomRect(window_menu, base_rect, localizedItem.computeNumUtf8CharsAndRewind() * window_menu.secondary_font->w);
```

Comment:
beware: computeNumUtf8CharsAndRewind will always run through the whole translated string to compute the number of glyphs - aka O(n)